beacon-chain/state/fieldtrie/field_trie_helpers.go

```diff
@@ -65,13 +66,36 @@ func validateElements(field types.FieldIndex, dataType types.DataType, elements
 // fieldConverters converts the corresponding field and the provided elements to the appropriate roots.
 func fieldConverters(field types.FieldIndex, indices []uint64, elements interface{}, convertAll bool) ([][32]byte, error) {
 	switch field {
-	case types.BlockRoots, types.StateRoots, types.RandaoMixes:
-		val, ok := elements.([][]byte)
+	case types.BlockRoots:
+		bVal, ok := elements.([][]byte)
```

Comment:
I think a type switch will be much cleaner than the nested if statements here. The same applies to the other fields too.